### ISSUE TITLE
Make flameshot process return exit code 1 when using `gui` and the screenshot is aborted

### ISSUE THREAD

**Issue by rodrigo-morales-1** [flameshot-org/flameshot#3755]
### Feature Description

Today I was writing a Python script that captures a region in the screen using `flameshot`. I noticed that flameshot returns the error code `0` when the screenshot is stored and when then screenshot is aborted.

Consider the Python script shown below which run `flameshot`:

```python
import os
import subprocess
import tempfile

temporary_directory = tempfile.mkdtemp()
temporary_file = os.path.join(temporary_directory, 'screenshot.png')
process_flameshot = subprocess.run(
  ['flameshot', 'gui', '--path', temporary_file],
  capture_output = True)

print(f"""stdout: '{process_flameshot.stdout.decode('utf-8').rstrip()}'""")
print(f"""stderr: '{process_flameshot.stderr.decode('utf-8').rstrip()}'""")
print(f"""returncode: {process_flameshot.returncode}""")
```

The following is the output when the user makes the selection and presses the button with a check mark (i.e. the screenshot is saved).

```text
$ python3 /tmp/a.py
stdout: ''
stderr: 'flameshot: info: Capture saved as /tmp/tmpcgi9smow/screenshot.png'
returncode: 0
```

The following is the output when the user presses the key ESC or the user makes the selection and presses the button with an X (i.e. the screenshot is aborted).

```text
$ python3 /tmp/a.py
stdout: ''
stderr: 'flameshot: info: Screenshot aborted.'
returncode: 0
```

Note that `returncode` equals `0` in both cases.

In the current version, if I wanted to check if the user aborted the screenshot using Python, I would need to look for the string `Screenshot aborted` in STDERR (as shown below):

```python
import os
import subprocess
import tempfile

temporary_directory = tempfile.mkdtemp()
temporary_file = os.path.join(temporary_directory, 'screenshot.png')
process_flameshot = subprocess.run(
  ['flameshot', 'gui', '--path', temporary_file],
  capture_output = True)

if "Screenshot aborted" in process_flameshot.stderr.decode('utf-8'):
  print("The screenshot wasn't saved.")
```

If the process returned 1, when the screenshot was aborted, then the code would look as shown below. To me the code below looks cleaner than the code above:

```python
import os
import subprocess
import tempfile

temporary_directory = tempfile.mkdtemp()
temporary_file = os.path.join(temporary_directory, 'screenshot.png')
process_flameshot = subprocess.run(
  ['flameshot', 'gui', '--path', temporary_file],
  capture_output = True)

if process_flameshot.returncode != 0:
  print("The screenshot wasn't saved.")
```

In summary, I think it would be great if the process returned `1` when `gui` is used and the screenshot is aborted. This would be convenient for people that use `flameshot` in scripts.

**Comment by mmahmoudian** [comment]
Thanks for the feature request. This is duplicate of #3321 and thankfully there is a PR (#3786) that is aiming to address this issue. I close this in favor of #3321 due to being duplicate. Thanks again.
